### ISSUE TITLE
beginproblem()  fix delicate behavior of \pgmlSetup

### ISSUE THREAD

**Issue by mgage** [openwebwork/webwork2#1066]
among other things this would fix a bug introduced in PR #1058

Here is the conversation after that PR was merged:
conf/snippets/ASimpleCombinedHeaderFile.pg 

@drgrice1
drgrice1 2 hours ago Member

Switching to PGML here seems to cause problems with hardcopy generation for problem sets that use this header file. The problem is that this causes the \pgmlSetup command to be used early in the tex file before it is defined.
@mgage
mgage 35 minutes ago Author Member

Good catch. I suspect that adding beginproblem() to header file will fix the hardcopy generation problem. It isn't completely natural since beginproblem() also outputs the
number of points a problem is worth. It's time to refactor beginproblem(). It's a hold over from WW1.9 when it was the only place we could initialize problems. It's performing too many functions.
 
it appears to be complicated. it would be worth asking @dpvc for advice.

This works fine:
```
loadMacros(
	"PG.pl",
	"PGbasicmacros.pl",
    "PGML.pl",
    "unionLists.pl",
	"PGcourse.pl",
);
#TEXT($BEGIN_ONE_COLUMN);
TEXT(beginproblem());
####################################################
#
# The item below printed out only when a hardcopy is made.
#
####################################################
```
If I comment out `TEXT(beginproblem())`  it breaks.
If I uncomment `TEXT($BEGIN_ONE_COLUMN)` and the matching $END_ONE_COLUMN
at the end of the file it also breaks.  I think this is because braces are introduced
into the TeX code which undefines \pgmlSetup where it is needed.  I seem to recall that this was a problem with scaffolding sections as well. It was important to enter certain header items in exactly the right order.

**Comment by mgage** [comment]
additional comment.  I think that beginproblem()  all alone without surrounding TEXT() will 
initialize pgmlSetup without printing points for the problem which is what is desired.

**Comment by mgage** [comment]
This version of the top of the header file works. The only caveat is that because beginproblem() is performing several functions it automatically assigns 0 pts to the header file and there does not seem to be an easy way to remove it.
```
loadMacros(
	"PG.pl",
	"PGbasicmacros.pl",
	#"source.pl", #uncommenting this will display "show source" button if that is permitted
	"PGcourse.pl",
);

TEXT($BEGIN_ONE_COLUMN);
loadMacros("PGML.pl");
TEXT(beginproblem() );

#This non-standard order allows PGML to work in hardcopy and using the one_column feature. 
...................
...................

TEXT($END_ONE_COLUMN);

ENDDOCUMENT(); 

```

**Comment by Alex-Jordan** [comment]
This is addressed by #1239.
